src/jupyter_contrib_nbextensions/nbextensions/execution_dependencies/execution_dependencies.js

```diff
@@ -0,0 +1,96 @@
+define([
+    'base/js/namespace',
+    'notebook/js/codecell'
+], function (
+    Jupyter,
+    codecell
+) {
+    "use strict";
+
+    var CodeCell = codecell.CodeCell;
+
+    return {
+        load_ipython_extension: function () {
+            console.log('[exec_deps] patching CodeCell.execute');
+            var orig_execute = codecell.CodeCell.prototype.execute;                                                 // get original cell execute function
+            CodeCell.prototype.execute = function (stop_on_error) {
+                var root_tags = this.metadata.tags || [];
+                if(root_tags != [] && root_tags.some(tag => /=>.*/.test(tag))) {                                    // if the root cell contains any dependencies, resolve dependency tree...
+                    var root_cell = this;
+                    var identified_cells = Jupyter.notebook.get_cells().filter(function (cell, idx, cells) {        // ...get all cells which have at least one id (these are the only ones we could have in deps)
+                        var tags = cell.metadata.tags || [];                        
+                        return (cell === root_cell || tags.some(tag => /#.*/.test(tag)));
+                    });
+
+                    console.log('Collect ids and dependencies...');
+                    var cell_map = {}  
+                    var dep_graph = {}
+                    identified_cells.forEach(function (cell) {                                                      // ...get all identified cells (the ones that have at least one #tag)
+                        var tags = cell.metadata.tags || [];
+                        var identities = tags.filter(tag => /#.*/.test(tag)).map(tag => tag.substring(1));          // ...get all identities and drop the #
+                        if(cell === root_cell && !tags.some(tag => /#.*/.test(tag))) {
```

Comment:
A single cell can have multiple identities, that is why it is called identities. I do not see why they would be required_ids. Below I just add an id, because the root cell does not need an id since it could be the one that is independent (the user is just working on that one and would not understand why it needs an id to run.).

Reply:
apologies, I misunderstood, for some reason I thought this was a list of dependencies' `id`s. No rename necessary :+1: